lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -1284,6 +1294,38 @@ prop_MultiPoolCertificateSequence_coverage mpcs = checkCoverage
     certificateSequences = getSinglePoolSequences mpcs
     poolCount = length certificateSequences
 
+
+-- | read . put == pure
+prop_putHeaderListHeader
+    :: DBLayer IO
+    -> [BlockHeader]
+    -> Positive Int
```

Comment:
Having thought about this, I would be inclined to change this to `NonNegative`.

Justification: the `listHeaders` function is not just defined on the set of positive values. In particular, it also has defined behaviour when `k == 0`.